crates/wascc-provider/src/lib.rs

```diff
@@ -650,28 +627,44 @@ impl kubelet::log::HandleFactory<tokio::fs::File> for LogHandleFactory {
 fn wascc_run(
     host: Arc<Mutex<WasccHost>>,
     data: Vec<u8>,
-    capabilities: &mut Vec<Capability>,
+    mut env: EnvVars,
     volumes: Vec<VolumeBinding>,
     log_path: &Path,
     status_recv: Receiver<ContainerStatus>,
+    port_assigned: i32,
 ) -> anyhow::Result<ContainerHandle<ActorHandle, LogHandleFactory>> {
+    let mut capabilities: Vec<Capability> = Vec::new();
     info!("sending actor to wascc host");
     let log_output = NamedTempFile::new_in(log_path)?;
     let mut logenv: HashMap<String, String> = HashMap::new();
     logenv.insert(
         LOG_PATH_KEY.to_string(),
         log_output.path().to_str().unwrap().to_owned(),
     );
-    capabilities.push(Capability {
-        name: LOG_CAPABILITY,
-        binding: None,
-        env: logenv,
-    });
 
     let load = Actor::from_bytes(data).map_err(|e| anyhow::anyhow!("Error loading WASM: {}", e))?;
     let pk = load.public_key();
 
-    if load.capabilities().contains(&FS_CAPABILITY.to_owned()) {
+    let actor_caps = load.capabilities();
+
+    if actor_caps.contains(&LOG_CAPABILITY.to_owned()) {
+        capabilities.push(Capability {
+            name: LOG_CAPABILITY,
+            binding: None,
+            env: logenv,
+        });
+    }
+
+    if actor_caps.contains(&HTTP_CAPABILITY.to_owned()) {
+        env.insert("PORT".to_string(), port_assigned.to_string());
+        capabilities.push(Capability {
+            name: HTTP_CAPABILITY,
+            binding: None,
+            env,
```

Comment:
So this is a little janky (not due to your code). For the actor to get the env variables that are assigned in Kubernetes, I believe those variables need to be assigned to the capability. So as it stands right now, the containers in the pod will only get the environment variables if they have the http capability.

Reply:
Can yo clarify what needs to be addressed here to fix this? Is there some test case I can use to verify this issue?

Reply:
The `env` hash map comes from the environment variables defined in the pod spec. We expect these variables to be set within the "container." These variables get passed to the actor when binding the provider. So if a provider doesn't support the http capability, it won't have the environment variables from the pod definition. You can test this by running an actor that prints out an environment variable specified in the pod spec, but doesn't have the http capability. You should see that the env variable is empty when it prints

Reply:
Okay. I think I see where you're coming from now. I'll see what I can do here, and see if I can write an integration test using @itowlson's wasmerciser.

Reply:
Reading the source code more closely, this seems intentional from a design standpoint. Each call to `bind_actor` contains a `HashMap<String, String>` that defines the configuration used between the actor and the capability. As far as I'm aware of, there is no concept of an environment variable in waSCC.

In other words, right now we're binding every environment variable in the pod manifest as configuration for the http server capability, and right now the [only value the http server capability cares about is PORT](https://github.com/wascc/http-server-provider/blob/7d538bb732f56e4e5dc1303c89cbce5515e72f7c/src/lib.rs#L83).

In other words, any environment variables you add to the pod manifest are currently ignored or unused. I'll open a new ticket to investigate how we should handle these going forward. Does that sound good to you?

Reply:
That sounds fine to me. We can tackle a solution in 0.5